src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -861,9 +861,17 @@ public void CloseDocumentCore(DocumentId documentId)
 
         protected override void ApplyDocumentTextChanged(DocumentId documentId, SourceText newText)
         {
-            EnsureEditableDocuments(documentId);
-            var hostDocument = GetHostDocument(documentId);
-            hostDocument.UpdateText(newText);
+            var filename = GetFilePath(documentId);
+            // When making project wide changes(i.e. change framework version),
+            // the TemporaryGeneratedFile_xxxx files under obj\Debug get removed and generated again.
+            // In that scenario, we won't find filename for the old documentId which was associated with
+            // the deleted TemporaryGeneratedFile_xxxx file, we just need to skip the following steps
+            if (filename != null)
```

Comment:
it wasn't clear to me that GetFilePath could return null.  We sholud rename that to TryGetFilePath.

Reply:
why not keep logic as before, but just have EnsureEditable return if was able to make the file editable.  if it wasn't, then don't do the get/update call.

Reply:
reverted